lib/segment/src/telemetry.rs

```diff
@@ -0,0 +1,228 @@
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+use std::sync::{Arc, Mutex};
+use std::time::{Duration, Instant};
+
+use serde::Serialize;
+
+use crate::types::{SegmentConfig, SegmentInfo};
+
+const AVG_DATASET_LEN: usize = 128;
+const SLIDING_WINDOW_LEN: usize = 8;
+
+pub trait Anonymize {
+    fn anonymize(&self) -> Self;
+}
+
+#[derive(Serialize, Clone)]
+pub struct SegmentTelemetry {
+    pub info: SegmentInfo,
+    pub config: SegmentConfig,
+    pub vector_index: VectorIndexTelemetry,
+    pub payload_field_indices: Vec<PayloadIndexTelemetry>,
+}
+
+#[derive(Serialize, Clone)]
+pub struct PayloadIndexTelemetry {}
+
+#[derive(Serialize, Clone)]
+pub struct VectorIndexTelemetry {
+    pub small_cardinality_searches: TelemetryOperationStatistics,
+    pub large_cardinality_searches: TelemetryOperationStatistics,
+    pub positive_check_cardinality_searches: TelemetryOperationStatistics,
+    pub negative_check_cardinality_searches: TelemetryOperationStatistics,
+}
+
+#[derive(Serialize, Clone, Default)]
+pub struct TelemetryOperationStatistics {
+    pub ok_count: usize,
+    pub fail_count: usize,
+    pub ok_avg_time: Duration,
```

Comment:
What do you think about using an histogram to record the distribution of latency.
This way we will know about really high max values and weird distribution.

Reply:
Maybe later. For now, we are using sliding window average and I think we gonna change it later

Reply:
We don't have an exact vision of measurements yet